public/test/mock-api/scenarios.ts

```diff
@@ -0,0 +1,25 @@
+import { HttpHandler } from 'msw';
+
+import { SILENCES_SCENARIOS } from './scenarios/alerting/silences';
+
+const ALL_SCENARIOS = {
+  ...SILENCES_SCENARIOS,
+};
+
+export type Scenario = keyof typeof ALL_SCENARIOS;
+
+/**
+ * Turns a list of scenario names into handlers that need to be `.use`'d by mock server/worker
+ * in order to make the scenarios "happen"
+ */
+export const getAdditionalScenarioHandlers = (scenarios: Scenario[]) => {
```

Comment:
This + setupScenarios/the worker query param logic should replace the `configure` methods that we have - this is a further level of abstraction that we can use to say "make these things happen"

Most of the configure methods should be easily moved into scenarios - though I found that doing the alertmanager config options was a bit verbose - we had to do something like this:, which is quite verbose. Any thoughts on alternative solutions?

```tsx
export const AM_SCENARIOS = {
  internalAlertmanager: [
    grafanaAlertingConfigurationStatusHandler({
      alertmanagersChoice: AlertmanagerChoice.Internal,
      numExternalAlertmanagers: 0,
    }),
  ],
  internalAlertmanagerAndOneExternal: [
    grafanaAlertingConfigurationStatusHandler({
      alertmanagersChoice: AlertmanagerChoice.Internal,
      numExternalAlertmanagers: 0,
    }),
  ],
  allAlertmanagers: [
    grafanaAlertingConfigurationStatusHandler({
      alertmanagersChoice: AlertmanagerChoice.All,
      numExternalAlertmanagers: 1,
    }),
  ],
  allAlertmanagersAndNoExternal: [
    grafanaAlertingConfigurationStatusHandler({
      alertmanagersChoice: AlertmanagerChoice.All,
      numExternalAlertmanagers: 0,
    }),
  ],
  noExternalAlertmanagers: [
    grafanaAlertingConfigurationStatusHandler({
      alertmanagersChoice: AlertmanagerChoice.External,
      numExternalAlertmanagers: 0,
    }),
  ],
  externalAlertmanagers: [
    grafanaAlertingConfigurationStatusHandler({
      alertmanagersChoice: AlertmanagerChoice.External,
      numExternalAlertmanagers: 1,
    }),
  ],
};
```